File: language/compiler/src/com/kubadziworski/parsing/visitor/expression/ValueExpressionVisitor.java
package com.kubadziworski.parsing.visitor.expression;

import jiva.parser.JivaBaseVisitor;
import jiva.parser.JivaParser;
import jiva.parser.JivaParser.ValueContext;
import com.kubadziworski.domain.node.expression.Value;
import com.kubadziworski.domain.type.Type;
import com.kubadziworski.util.TypeResolver;
import org.antlr.v4.runtime.misc.NotNull;

public class ValueExpressionVisitor extends JivaBaseVisitor<Value> {

    @Override
    public Value visitValue(@NotNull ValueContext ctx) {
        String value = ctx.getText();
        Type type = TypeResolver.getFromValue(ctx);
        return new Value(type, value);
    }
}

File: language/compiler/src/parsing/visitor/expression/ValueExpressionVisitor.kt
package jiva.parsing.visitor.expression

import jiva.domain.node.expression.Value
import jiva.util.TypeResolver
import jiva.parser.JivaBaseVisitor
import jiva.parser.JivaParser.ValueContext

class ValueExpressionVisitor : JivaBaseVisitor<Value>() {

    override fun visitValue(ctx: ValueContext): Value {
        val value = ctx.text
        val type = TypeResolver.getFromValue(ctx)
        return Value(type, value)
    }

}
